teuthology/orchestra/remote_ops.py

```diff
@@ -0,0 +1,295 @@
+"""
+Contains methods to run standard operations (like creating/deleting files) on
+remote machines.
+
+These methods were originally part of orchestra.remote.Remote. The reason for
+moving these methods from Remote to here is that applications that use
+teuthology for testing usually have programs that can run tests locally on a
+single node machine for development work (for example, vstart_runner.py in
+case of Ceph). These programs can import and reuse these methods without
+having to deal SSH stuff.
+
+To use these methods, inherit the class here and implement "run()" method in
+the subclass.
+"""
+
+from errno import ENOENT as errno_ENOENT
+from io import BytesIO
+from io import StringIO
+from os.path import dirname as os_path_dirname
+
+from teuthology.exceptions import CommandFailedError
+from teuthology.lock.query import is_vm
+from teuthology.orchestra.opsys import OS
+
+
+class NonTransferRemoteOps(object):
```

Comment:
The name still irks me. These are all just shell commands. Something like `RemoteShellOps` would be appropriate.

Reply:
Now, I think, we should move that code back to Remote :-D seriously, instead maybe it is better to moveout just few methods Remote.sh to RemoteShell as a base class for remote?

Reply:
Well... Looks like we need just base Runner class which can be overridden by RemoteRunner and LocalRunner (I don't know if this is an appropriate name for vstart use case)

Reply:
I've changed `NonTransferRemoteOps` to `RemoteShell` and `remote_ops.py` to `remote_shell.py` in latest push.

Reply:
BTW, I still feel that moving `RemoteShell` to a separate module is a bit of an overkill. It's best to keep it in `remote.py`.

Reply:
@kshtsk 
> Well... Looks like we need just base Runner class which can be overridden by RemoteRunner and LocalRunner (I don't know if this is an appropriate name for vstart use case)

Programs like `vstart_runner.py` don't need those SSH bits that current `remote.Remote` contains .

Reply:
> BTW, I still feel that moving `RemoteShell` to a separate module is a bit of an overkill. It's best to keep it in `remote.py`.

Agree.

What about to have two classes RemoteShell which will include only ssh run and sh, and have LocalShell, both based on AbstractShell. The Remote class should use either of the RemoteShell or LocalShell to construct.

Reply:
@kshtsk Just separating `sh()` and `run()` won't separate methods that depend on SSH from methods that don't which is the goal of this PR. Or do you mean to make this a further change on the top of the current patch?

Perhaps, I am not clear about what you want to say. In that case, can we chat about this on IRC? What's you nickname? Mine is rishabh.

Reply:
FYI: I've moved contents remote_shell.py as it is to remote.py.